### PR TITLE
CDAP-2069 adding a filter transform that lets the user provide

### PR DESCRIPTION
**Pull request by albertshau**
a javascript snippet that determines whether or not a record
should be filtered or not.

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DT16

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/templates/etl/transforms/ScriptFilterTransformTest.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import co.cask.cdap.templates.etl.common.MockEmitter;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ */
+@SuppressWarnings("unchecked")
+public class ScriptFilterTransformTest {
```
Comment (gokulavasan) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/templates/etl/transforms/ScriptFilterTransformTest.java]: May be we can add couple of tests to test the failure scenarios - invalid script and invalid filter condition?

**Comment by gokulavasan** [comment]
Comment about testing error conditions. Rest LGTM :+1:

File: cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/templates/etl/transforms/ScriptFilterTransformTest.java (hunk 1 above)
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/templates/etl/transforms/ScriptFilterTransformTest.java]: sure will do

**Pull request merged**